src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketGitSCMBuilder.java

```diff
@@ -0,0 +1,272 @@
+                BitbucketRepositoryType.GIT,
+                protocol,
+                protocolPortOverride,
+                repoOwner,
+                repository));
+        AbstractBitbucketEndpoint endpoint =
+                BitbucketEndpointConfiguration.get().findEndpoint(scmSource.getServerUrl());
+        if (endpoint == null) {
+            endpoint = new BitbucketServerEndpoint(null, scmSource.getServerUrl(), false, null);
+        }
+        withBrowser(new BitbucketWeb(
+                endpoint.getRepositoryUrl(
+                        repoOwner,
+                        repository
+                )));
+
+        // now, if we have to build a merge commit, let's ensure we build the merge commit!
+        SCMRevision r = revision();
+        if (h instanceof PullRequestSCMHead) {
+            PullRequestSCMHead head = (PullRequestSCMHead) h;
+            if (head.getCheckoutStrategy() == ChangeRequestCheckoutStrategy.MERGE) {
+                String name = head.getTarget().getName();
+                String localName = head.getBranchName().equals(name) ? "upstream-" + name : name;
+
+                String remoteName = remoteName().equals("upstream") ? "upstream-upstream" : "upstream";
+                withAdditionalRemote(remoteName,
+                        bitbucket.getRepositoryUri(
+                                BitbucketRepositoryType.GIT,
+                                protocol,
+                                protocolPortOverride,
+                                scmSource().getRepoOwner(),
+                                scmSource().getRepository()),
+                        "+refs/heads/" + localName + ":refs/remotes/@{remote}/" + name);
+                if ((r instanceof PullRequestSCMRevision)
+                        && ((PullRequestSCMRevision) r).getTarget() instanceof AbstractGitSCMSource.SCMRevisionImpl) {
+                    withExtension(new MergeWithGitSCMExtension("remotes/" + remoteName + "/" + localName,
+                            ((AbstractGitSCMSource.SCMRevisionImpl) ((PullRequestSCMRevision) r).getTarget())
+                                    .getHash()));
+                } else {
+                    withExtension(new MergeWithGitSCMExtension("remotes/" + remoteName + "/" + localName, null));
+                }
+            }
+        }
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @NonNull
+    @Override
+    public GitSCM build() {
+        withBitbucketRemote();
+        SCMHead h = head();
+        SCMRevision r = revision();
+        try {
+            if (h instanceof PullRequestSCMHead) {
+                withHead(new SCMHead(((PullRequestSCMHead) h).getBranchName()));
+                if (r instanceof PullRequestSCMRevision) {
+                    withRevision(((PullRequestSCMRevision) r).getPull());
+                }
+            }
+            return super.build();
+        } finally {
+            withHead(h);
+            withRevision(r);
```

Comment:
Inside the try we swap out the Pull request head and revision for PRs... this just puts them back so that repeated calls will still get the PR directed changes